src/os/README.md

```diff
@@ -104,3 +106,86 @@ inspiration for several decisions in this library.
 - that said, if an operation is extremely efficient and almost always wanted,
   consider doing it natively, automatically. For example, ints in HSL
   `sockaddr` are always in host byte order, not network byte order.
+
+## Hack representation of C data types 
+### Cheat sheet
+
+| C type | Hack type |
+|---|---|
+| Pointers to referential transparent C `struct`s | `vec` |
+| Other C handles or pointers | `HH\Lib\OS\FileDescriptor` or other wrapper classes |
+
+### Default encoding for handles or pointers
+
+The underlying C API commonly exposes some data type as handlers or pointers,
+e.g. `int filedes`, along with functions to create, manipulate, and destroy
+them. By default, handlers and pointers should be wrapped in HSL as Hack
+classes.
+
+### Encoding referential transparent C pointers
+
+However, as an optimization, pointers to referential transparent C `struct`s can
+be encoded as a garbage-collectable mirror in Hack, which creates and destroys
+the underlying C data type on demand. 
```

Comment:
which creates the underlying C data type on demand, and destroys it when explicitly cleaned up by a dedicated function call, or at some later undetermined time which should not be directly observable by the same request

Reply:
Updated